scripts/pythran-config

```diff
@@ -21,14 +26,28 @@ args = parser.parse_args(sys.argv[1:])
 
 output = []
 
+extension = make_extension()
+
 if args.compiler:
-    output.append(tc.default_compiler())
+    output.append(os.environ.get('CXX', 'c++'))
 
 if args.cflags:
-    output.extend(tc.cppflags() + tc.cxxflags())
+    def fmt_define(define):
+        name, value = define
+        if value is None:
+            return '-D' + name
+        else:
+            return '-D' + name + '=' + value
+    output.extend(fmt_define(define) for define in extension['define_macros'])
+    output.extend(('-I' + include) for include in extension['include_dirs'])
+    output.append('-I' + numpy.get_include())
+    output.append('-I' + distutils.sysconfig.get_python_inc())
```

Comment:
On Fri, May 22, 2015 at 07:16:10AM -0700, pbrunet wrote:
> >  
> >  if args.cflags:
> > -    output.extend(tc.cppflags() + tc.cxxflags())
> > +    def fmt_define(define):
> > +        name, value = define
> > +        if value is None:
> > +            return '-D' + name
> > +        else:
> > +            return '-D' + name + '=' + value
> > +    output.extend(fmt_define(define) for define in extension['define_macros'])
> > +    output.extend(('-I' + include) for include in extension['include_dirs'])
> > +    output.append('-I' + numpy.get_include())
> > +    output.append('-I' + distutils.sysconfig.get_python_inc())
> 
> It is not the default for distutils.Extension?

It is, but it's internal to distutils and I cannot manage to extract the
info from the Extension (it's probably built in the build_ext somewhere)

I hate this piece of code, but cannot find a bettre way (I tried!)